metriq_gym/run.py

```diff
@@ -126,7 +181,9 @@ def main() -> int:
     if args.action == "dispatch":
         dispatch_job(args, job_manager)
     elif args.action == "poll":
-        poll_job(args, job_manager)
+        poll_job(args, job_manager, False)
+    elif args.action == "upload":
+        poll_job(args, job_manager, True)
```

Comment:
Should we maybe architect this a bit differently such that if `upload` is the action then the`upload_job` function is called? Calling `poll_job` with a different bool feels like it might be confusing for refactoring down the road and breaks the current arg.action checking.

Reply:
The difference in `poll_job` is 4 lines for one optional branch, though. We do want to reuse the code, for that reason, I think, but the function could be named differently, like `poll_with_upload()`.